packages/backend/src/spsp/service.ts

```diff
@@ -0,0 +1,111 @@
+import { BaseService } from '../shared/baseService'
+import { AppContext } from '../app'
+import { AccountService } from '../account/service'
+import { UserService } from '../user/service'
+import { validate } from 'uuid'
+import base64url from 'base64url'
+import { StreamServer } from '@interledger/stream-receiver'
+
+const CONTENT_TYPE_V4 = 'application/spsp4+json'
+
+export interface SPSPService {
+  GETPayEndpoint(ctx: AppContext): Promise<void>
+  streamServer(): Promise<StreamServer>
+}
+
+interface ServiceDependencies extends BaseService {
+  accountService: AccountService
+  userService: UserService
+  streamServer: StreamServer
+}
+
+export async function createSPSPService({
+  logger,
+  accountService,
+  userService,
+  streamServer
+}: ServiceDependencies): Promise<SPSPService> {
+  const log = logger.child({
+    service: 'UserService'
+  })
+
+  const deps: ServiceDependencies = {
+    logger: log,
+    accountService: accountService,
+    userService: userService,
+    streamServer: streamServer
+  }
+  return {
+    GETPayEndpoint: (ctx) => getPay(deps, ctx),
+    streamServer: () => getStreamServer(deps)
```

Comment:
Would it be better to make the `streamServer` a service/property attached to the container, so a getter isn't needed here? It's awkward for the SPSP service to provide access to an object that it neither creates nor modifies.

Reply:
Yes that makes sense.